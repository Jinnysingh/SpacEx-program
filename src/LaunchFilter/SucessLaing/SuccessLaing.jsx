import React from "react";

function SuccessLaing(props) {
  return (
      <div className="launch-sec">
      <div className="title">
      <u>Successful Laing</u>
      </div>
      {props.launchFilters.map((items) => (
        <button type="button" className="btn btn-success text-dark launch-btn active-btn" onClick={(event) => props.handleFilters(event, "land_success")}>
          {items}
        </button>
      ))}
    </div>
  );
}

export default SuccessLaing;
