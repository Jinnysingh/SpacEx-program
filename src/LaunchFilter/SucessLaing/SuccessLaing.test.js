import {mount} from "enzyme"
import toJSON from "enzyme-to-json"
import React from "react"
import SuccessLaing from "./SuccessLaing.jsx"

let launchFilters = ["true", "false"]

describe('<SuccessLaing />', () => {
    const wrapper = mount(
      <SuccessLaing launchFilters={launchFilters} />
    );

test("testing the filters", ()=>{
  expect(launchFilters).not.toBe(0);
  expect(toJSON(wrapper)).toMatchSnapshot();
})
})