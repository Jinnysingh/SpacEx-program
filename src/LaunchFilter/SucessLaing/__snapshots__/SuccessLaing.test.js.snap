// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessLaing /> testing the filters 1`] = `
<SuccessLaing
  launchFilters={
    Array [
      "true",
      "false",
    ]
  }
>
  <div
    className="launch-sec"
  >
    <div>
      <u>
        Successful Laing
      </u>
    </div>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      true
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      false
    </button>
  </div>
</SuccessLaing>
`;

exports[`<launchYears /> testing the filters 1`] = `
<SuccessLaing
  launchFilters={
    Array [
      "true",
      "false",
    ]
  }
>
  <div
    className="launch-sec"
  >
    <div>
      <u>
        Successful Laing
      </u>
    </div>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      true
    </button>
    <button
      className="btn btn-success text-dark launch-btn active-btn"
      onClick={[Function]}
      type="button"
    >
      false
    </button>
  </div>
</SuccessLaing>
`;
